tls/s2n_psk.c

```diff
@@ -291,9 +317,30 @@ int s2n_offered_psk_list_choose_psk(struct s2n_offered_psk_list *psk_list, struc
         psk_params->chosen_psk = NULL;
         return S2N_SUCCESS;
     }
-
-    POSIX_GUARD_RESULT(s2n_match_psk_identity(&psk_params->psk_list, &psk->identity, &psk_params->chosen_psk));
-    psk_params->chosen_psk_wire_index = psk->wire_index;
+    
+    if(psk_params->type == S2N_PSK_TYPE_EXTERNAL) {
+        POSIX_GUARD_RESULT(s2n_match_psk_identity(&psk_params->psk_list, &psk->identity, &psk_params->chosen_psk));
+        psk_params->chosen_psk_wire_index = psk->wire_index;
```

Comment:
You set the wire index for both cases, so that could be done outside of the if/else